samples/gen-go-db/server/db/dynamodb/deployment.go

```diff
@@ -0,0 +1,557 @@
+		Item:      data,
+	})
+	return err
+}
+
+func (t DeploymentTable) getDeployment(ctx context.Context, environment string, application string, version string) (*models.Deployment, error) {
+	key, err := dynamodbattribute.MarshalMap(ddbDeploymentPrimaryKey{
+		EnvApp:  fmt.Sprintf("%s--%s", environment, application),
+		Version: version,
+	})
+	if err != nil {
+		return nil, err
+	}
+	res, err := t.DynamoDBAPI.GetItemWithContext(ctx, &dynamodb.GetItemInput{
+		Key:       key,
+		TableName: aws.String(t.name()),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	if len(res.Item) == 0 {
+		return nil, db.ErrDeploymentNotFound{
+			Environment: environment,
+			Application: application,
+			Version:     version,
+		}
+	}
+
+	var m models.Deployment
+	if err := decodeDeployment(res.Item, &m); err != nil {
+		return nil, err
+	}
+
+	return &m, nil
+}
+
+func (t DeploymentTable) getDeploymentsByEnvAppAndVersion(ctx context.Context, input db.GetDeploymentsByEnvAppAndVersionInput, fn func(m *models.Deployment, lastDeployment bool) bool) error {
+	if input.VersionStartingAt != nil && input.StartingAfter != nil {
+		return fmt.Errorf("Can specify only one of input.VersionStartingAt or input.StartingAfter")
+	}
+	queryInput := &dynamodb.QueryInput{
+		TableName: aws.String(t.name()),
+		ExpressionAttributeNames: map[string]*string{
+			"#ENVAPP": aws.String("envApp"),
+		},
+		ExpressionAttributeValues: map[string]*dynamodb.AttributeValue{
+			":envApp": &dynamodb.AttributeValue{
+				S: aws.String(fmt.Sprintf("%s--%s", input.Environment, input.Application)),
+			},
+		},
+		ScanIndexForward: aws.Bool(!input.Descending),
+		ConsistentRead:   aws.Bool(!input.DisableConsistentRead),
+	}
+	if input.Limit != nil {
+		queryInput.Limit = input.Limit
+	}
+	if input.VersionStartingAt == nil {
+		queryInput.KeyConditionExpression = aws.String("#ENVAPP = :envApp")
+	} else {
+		queryInput.ExpressionAttributeNames["#VERSION"] = aws.String("version")
+		queryInput.ExpressionAttributeValues[":version"] = &dynamodb.AttributeValue{
+			S: aws.String(*input.VersionStartingAt),
+		}
+		if input.Descending {
+			queryInput.KeyConditionExpression = aws.String("#ENVAPP = :envApp AND #VERSION <= :version")
+		} else {
+			queryInput.KeyConditionExpression = aws.String("#ENVAPP = :envApp AND #VERSION >= :version")
+		}
+	}
+	if input.StartingAfter != nil {
+		queryInput.ExclusiveStartKey = map[string]*dynamodb.AttributeValue{
+			"version": &dynamodb.AttributeValue{
+				S: aws.String(input.StartingAfter.Version),
+			},
+			"envApp": &dynamodb.AttributeValue{
+				S: aws.String(fmt.Sprintf("%s--%s", input.StartingAfter.Environment, input.StartingAfter.Application)),
+			},
+		}
+	}
+
+	var decodeErr error
+	var items []models.Deployment
+	pageFn := func(queryOutput *dynamodb.QueryOutput, lastPage bool) bool {
+		if len(queryOutput.Items) == 0 {
+			return false
+		}
+		items, decodeErr = decodeDeployments(queryOutput.Items)
+		if decodeErr != nil {
+			return false
+		}
```

Comment:
I get a little wierded out by reusing the items array here and putting it in the outer scope, maybe consider

```
		items, err := decodeDeployments(queryOutput.Items)
		if err != nil {
			pageFnErr = err // pageFnErr might be a better name?
			return false
		}
```